include/ignition/gazebo/gui/GuiEvents.hh

```diff
@@ -176,6 +179,44 @@ namespace events
     /// \brief True if a transform mode is active.
     private: bool tranformModeActive;
   };
+
+  /// \brief Event that notifies an entity is to be added to the model editor
+  class ModelEditorAddEntity : public QEvent
+  {
+    /// \brief Constructor
+    /// \param[in] _tranformModeActive is the transform control mode active
+    public: explicit ModelEditorAddEntity(QString _entity, QString _type,
+                ignition::gazebo::Entity _parent) :
+      QEvent(kType), entity(_entity), type(_type), parent(_parent)
+    {
+    }
+
+    /// \brief Get the entity to add
+    public: QString Entity() const
+    {
+      return this->entity;
+    }
+
+    /// \brief Get the entity type
+    public: QString EntityType() const
+    {
+      return this->type;
+    }
+
+    /// \brief Get the parent entity to add the entity to
+    public: ignition::gazebo::Entity ParentEntity() const
+    {
+      return this->parent;
+    }
+
+    static const QEvent::Type kType = QEvent::Type(QEvent::User + 7);
+
+    public : QMap<QString, QString> data;
+    private: QString entity;
+    private: QString type;
+    private: ignition::gazebo::Entity parent;
```

Comment:
Can we PIMPL this class? There's a couple of events above to look as examples.

Reply:
I'll put this on my todo list after this PR.

Reply:
Pimplized in a9f9443

Reply:
https://github.com/ignitionrobotics/ign-gazebo/pull/1233